test/e2e/nsx_subnet_test.go

```diff
@@ -602,3 +699,570 @@ func SubnetValidate(t *testing.T) {
+	require.Equal(t, originalVPCName, finalSubnet.Spec.VPCName, "VPCName should remain unchanged")
+	require.Equal(t, originalEnableVLANExtension, finalSubnet.Spec.AdvancedConfig.EnableVLANExtension,
+		"EnableVLANExtension should remain unchanged")
+
+	log.Info("Verified that shared subnet properties cannot be updated", "subnet", subnet2.Name)
+}
+
+// NormalSubnetManagedByNSXOp tests that when a namespace user creates a normal Subnet CR in the namespace,
+// its status.shared is false, and when getting the subnet from NSX, its tags contain nsx/managed-by:nsx-op
+func NormalSubnetManagedByNSXOp(t *testing.T) {
+	// Create a normal subnet in the namespace
+	normalSubnet := &v1alpha1.Subnet{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "normal-subnet",
+			Namespace: ns1,
+		},
+		Spec: v1alpha1.SubnetSpec{
+			AccessMode: v1alpha1.AccessMode(v1alpha1.AccessModePrivate),
+			AdvancedConfig: v1alpha1.SubnetAdvancedConfig{
+				StaticIPAllocation: v1alpha1.StaticIPAllocation{
+					Enabled: common.Bool(true),
+				},
+			},
+			SubnetDHCPConfig: v1alpha1.SubnetDHCPConfig{
+				Mode: v1alpha1.DHCPConfigMode(v1alpha1.DHCPConfigModeDeactivated),
+			},
+		},
+	}
+
+	// Create the subnet and verify it's created successfully
+	createdSubnet := createSubnetWithCheck(t, normalSubnet)
+	require.NotNil(t, createdSubnet, "Failed to create normal subnet in namespace %s", ns1)
+
+	// Verify that status.shared is false
+	require.False(t, createdSubnet.Status.Shared, "Normal subnet should have status.shared=false")
+	log.Info("Verified normal subnet has status.shared=false", "subnet", createdSubnet.Name)
+
+	// Get the subnet from NSX using the subnet UID
+	subnetCRUID := string(createdSubnet.UID)
+	nsxSubnets := testData.fetchSubnetBySubnetUID(t, subnetCRUID)
+	require.Equal(t, 1, len(nsxSubnets), "Expected to find exactly one NSX subnet for the created subnet")
+
+	// Check that the NSX subnet has the nsx/managed-by:nsx-op tag
+	nsxSubnet := nsxSubnets[0]
+	found := false
+	for _, tag := range nsxSubnet.Tags {
+		if *tag.Scope == common.TagScopeManagedBy && *tag.Tag == common.AutoCreatedTagValue {
+			found = true
+			break
+		}
+	}
+	require.True(t, found, "NSX subnet should have the tag %s:%s", common.TagScopeManagedBy, common.AutoCreatedTagValue)
+	log.Info("Verified NSX subnet has the correct managed-by tag", "subnet", createdSubnet.Name)
+
+	// Clean up - delete the subnet
+	err := testData.crdClientset.CrdV1alpha1().Subnets(ns1).Delete(context.TODO(), normalSubnet.Name, v1.DeleteOptions{})
+	require.NoError(t, err, "Failed to delete normal subnet %s", normalSubnet.Name)
+
+	// Wait for the subnet to be deleted
+	err = wait.PollUntilContextTimeout(context.TODO(), 1*time.Second, 30*time.Second, false, func(ctx context.Context) (bool, error) {
+		_, err := testData.crdClientset.CrdV1alpha1().Subnets(ns1).Get(context.TODO(), normalSubnet.Name, v1.GetOptions{})
+		return errors.IsNotFound(err), nil
+	})
```

Comment:
Maybe check nsx Subnet instead of k8s Subnet, as the deletion of k8s Subnet is actually controlled by k8s instead of nsx operator?

Reply:
fixed